sim-cli/src/main.rs

```diff
@@ -42,9 +48,12 @@ fn deserialize_f64_greater_than_zero(x: String) -> Result<f64, String> {
 #[derive(Parser)]
 #[command(version, about)]
 struct Cli {
+    /// Path to a directory containing simulation files, and where simulation results will be stored
+    #[clap(long, short, env = "SIM_LN_DATA_DIR", default_value = DEFAULT_DATA_DIR)]
+    data_dir: PathBuf,
     /// Path to the simulation file to be used by the simulator
-    #[clap(index = 1)]
-    sim_file: PathBuf,
+    #[clap(long, short, default_value = DEFAULT_SIM_FILE)]
+    sim_file: String,
```

Comment:
[meta] We do have a few people using this in the wild, and this will be a breaking change for them :( 

I think this is _okay_, but we do need to eventually draw the line for this type of change because it can be super annoying when your tooling breaks every time you upgrade.

Reply:
Agreed on goal to limit breaking changes on public api and experience. In this case, I'm hoping `sim.json` as `DEFAULT_SIM_FILE` coupled with the prompt to select a file from some default or configured data dir more than makes up for breaking current behavior. We cannot preserve a positional `sim_file` arg because this path requirement would likely to conflict with path to data dir

Reply:
Yeah, agree it's nice enough UX improvement to justify the break in this case :)